### PR TITLE
qa/tasks/vstart_runner.py: add def mountpoint parameter

### PR DESCRIPTION
**Pull request by lxbsz**
INFO:__main__:test_replay_create (tasks.cephfs.test_cap_flush.TestCapFlush) ... ERROR
INFO:__main__:Stopped test: test_replay_create (tasks.cephfs.test_cap_flush.TestCapFlush) in 22.251901s
INFO:__main__:
INFO:__main__:======================================================================
INFO:__main__:ERROR: test_replay_create (tasks.cephfs.test_cap_flush.TestCapFlush)
INFO:__main__:----------------------------------------------------------------------
INFO:__main__:Traceback (most recent call last):
INFO:__main__:  File "/data/ceph/qa/tasks/cephfs/cephfs_test_case.py", line 145, in setUp
INFO:__main__:    self.mounts[i].mount_wait()
INFO:__main__:  File "/data/ceph/qa/tasks/cephfs/mount.py", line 420, in mount_wait
INFO:__main__:    mount_options=mount_options)
INFO:__main__:TypeError: mount() got an unexpected keyword argument 'mountpoint'
INFO:__main__:
INFO:__main__:----------------------------------------------------------------------
INFO:__main__:Ran 1 test in 22.253s
INFO:__main__:
INFO:__main__:FAILED (errors=1)
INFO:__main__:

Fixes: https://tracker.ceph.com/issues/45300
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/vstart_runner.py [hunk 1]
```diff
@@ -583,7 +583,7 @@ def _asok_path(self):
         log.info("I think my launching pid was {0}".format(self.fuse_daemon.subproc.pid))
         return path
 
-    def mount(self, mount_path=None, mount_fs_name=None, mount_options=[]):
+    def mount(self, mount_path=None, mount_fs_name=None, mountpoint=None, mount_options=[]):
```
Comment (jtlayton) [inline qa/tasks/vstart_runner.py:586]: This argument is unused in the function. Should it override the usage of self.mountpoint?

Comment (lxbsz) [reply qa/tasks/vstart_runner.py:586]: Yeah, you are right, and fixed it now.

File: qa/tasks/vstart_runner.py [hunk 2]
```diff
@@ -583,7 +583,9 @@ def _asok_path(self):
...
+        if mountpoint is not None:
+            self.mountpoint = mountpoint
```
Comment (jtlayton) [inline qa/tasks/vstart_runner.py:588]: While you're in here... Should this function return an error if self.mounted is already true? It seems like you could call this once with one mountpoint and then again with another, and it would lose track of the first one. Granted I know this is mostly for testing, so I'm not clear on how robust it needs to be...

Comment (lxbsz) [reply qa/tasks/vstart_runner.py:588]: Checked the test cases, when the mountpoint is specified with a new path, and if it umounted and reused in other followed test cases, the old mountpoint path will be lost.
We should fix them all by restoring the default mount point when umounting.

File: qa/tasks/vstart_runner.py (hunk 1 above)
Comment (batrick) [reply qa/tasks/vstart_runner.py:592]: I think it's better to not set `self.mountpoint` here. fuse/kernel mount should not be doing redundant things.

Better would be:

`def mount(self, ..., **kwargs):`

so that it ignores any args it does not understand.

File: qa/tasks/cephfs/kernel_mount.py [hunk 3]
```diff
@@ -94,6 +94,7 @@ def umount(self, force=False):
 
         self.mounted = False
         self.cleanup()
+        super(KernelMount, self).umount()
```
Comment (batrick) [inline qa/tasks/cephfs/kernel_mount.py:97]: This looks unrelated to the PR. Please have a separate commit explaining.

File: qa/tasks/cephfs/mount.py [hunk 4]
```diff
@@ -60,7 +60,6 @@ def mountpoint(self, path):
         if not isinstance(path, str):
             raise RuntimeError('path should be of str type.')
         self._mountpoint = path
-        self._parse_netns_name()
```
Comment (batrick) [inline qa/tasks/cephfs/mount.py:63]: This looks unrelated to the PR. Please have a separate commit explaining.

File: qa/tasks/cephfs/mount.py [hunk 5]
```diff
@@ -341,7 +340,7 @@ def mount_wait(self, mount_path=None, mount_fs_name=None, mountpoint=None, mount
         self.wait_until_mounted()
 
     def umount(self):
-        raise NotImplementedError()
+        self._mountpoint = None
```
Comment (batrick) [inline qa/tasks/cephfs/mount.py:343]: NAK

File: qa/tasks/cephfs/kernel_mount.py (hunk 3 above)
Comment (lxbsz) [reply qa/tasks/cephfs/kernel_mount.py]: Will add the kwargs instead and remove the unrelated code in this PR.
